lib/internal/encoding.js

```diff
@@ -32,6 +32,21 @@ function lazyBuffer() {
   return Buffer;
 }
 
+function validateEncoder(obj) {
+  if (obj == null || obj[kEncoder] !== true)
+    throw new errors.TypeError('ERR_INVALID_THIS', 'TextEncoder');
+}
+
+function validateDecoder(obj) {
+  if (obj == null || obj[kDecoder] !== true)
+    throw new errors.TypeError('ERR_INVALID_THIS', 'TextDecoder');
+}
+
+function validateArgument(prop, expected, propName) {
+  if (typeof prop !== expected && !isArrayBufferView(prop))
```

Comment:
This is not only a extraction but it changes the validation itself as it also validates that the input is not an `ArrayBufferView`. I would like to keep it as is.